src/v2/Components/v2/ArtworkFilter/__tests__/ArtworkFilterContext.jest.tsx

```diff
@@ -129,28 +95,141 @@ describe("ArtworkFilterContext", () => {
+      beforeEach(() => {
+        let filters = {
           ...initialArtworkFilterState,
-          acquireable: true,
-          at_auction: true,
-        },
+          medium: "painting",
+        }
+
+        // init a 'real' filter state
+        getWrapper({ filters })
+
+        // copy that to a 'staged' filter state
+        act(() => {
+          context.setStagedFilters(filters)
+        })
       })
-      expect(context.filters.acquireable).toEqual(true)
-      expect(context.filters.atAuction).toEqual(true)
 
-      act(() => {
-        context.resetFilters()
-        setTimeout(() => {
-          expect(context.filters).toEqual(initialArtworkFilterState)
+      describe("when not in staged mode", () => {
+        beforeEach(() => {
+          act(() => {
+            context.setShouldStageFilterChanges(false)
+          })
+        })
+
+        test("#setFilter sets only 'real' filters", () => {
+          act(() => {
+            context.setFilter("medium", "jewelry")
+          })
+          expect(context.filters.medium).toEqual("jewelry")
+          expect(context.stagedFilters.medium).not.toEqual("jewelry")
+        })
+
+        test("#unsetFilter unsets only 'real' filters", () => {
+          act(() => {
+            context.unsetFilter("medium")
+          })
+          expect(context.filters.medium).not.toEqual("painting")
+          expect(context.stagedFilters.medium).toEqual("painting")
+        })
+
+        test("#resetFilters resets only 'real' filters", () => {
+          act(() => {
+            context.resetFilters()
+          })
+          expect(context.filters).toEqual({
+            ...initialArtworkFilterState,
+            reset: true,
+          })
+          expect(context.stagedFilters).not.toEqual({
+            ...initialArtworkFilterState,
+            reset: true,
+          })
+        })
+
+        test("#currentlySelectedFilters returns the 'real' filters", () => {
+          act(() => {
+            context.setFilter("medium", "design")
+          })
+          expect(context.currentlySelectedFilters()).toEqual(context.filters)
+        })
+      })
+
+      describe("when in staged mode", () => {
+        beforeEach(() => {
+          act(() => {
+            context.setShouldStageFilterChanges(true)
+          })
+        })
+
+        test("#setFilter sets only staged filters", () => {
+          act(() => {
+            context.setFilter("medium", "jewelry")
+          })
+          expect(context.filters.medium).not.toEqual("jewelry")
+          expect(context.stagedFilters.medium).toEqual("jewelry")
+        })
+
+        test("#unsetFilter unsets only staged filters", () => {
+          act(() => {
+            context.unsetFilter("medium")
+          })
+          expect(context.filters.medium).toEqual("painting")
+          expect(context.stagedFilters.medium).not.toEqual("painting")
+        })
+
+        test("#resetFilters resets only staged filters", () => {
```

Comment:
I _think_ we'll want resetting the filters to reset both applied + staged filters (otherwise on mobile web there would be no way to clear out applied filters, right?).

Was trying to follow the control flow in the code above (and not just referring to this spec name)-- it seems like it's only resetting staged filters but I totally could be misreading.

For context, on iOS "Clear All" does reset _both_ applied + staged filters (and the "diff" is staged, so you can still close out of the change entirely).

Reply:
<img height=300 src="https://user-images.githubusercontent.com/140521/83833054-0918c400-a6b9-11ea-8d32-64fb9153c123.gif" />

Right now, pressing **Clear all** immediately resets all staged filters but doesn't apply that change until you actually hit the **Apply** button. At that point the (now empty) staged filters overwrite the real filters.

Do we want to change the **Clear all** button such that it _immediately_ closes the modal and applies, without waiting for the user to hit **Apply**?

(Took a second look at the comp, and didn't see advice one way or the other.)

Reply:
Re-reading your comment and looking at iOS collections — I think the Clear/Cancel/Apply cycle actually works the same here as on iOS.

But the **Apply (n)** count differs.

Lmk if you agree and if we should follow up with just that count change.

Reply:
Just to make sure we're on the same page, what happens in the following scenarios?:
1. Visit the /collect page
2. Open the filter sheet
3. Filter by "Painting" and click "Apply"
4. Open the filter sheet
5.

Reply:
And yeah-- it looks like the count also differs. That seems like something we can ticket as a follow-up!

Reply:
> (First scenario)

Works as you described ✔️ 

> (Second scenario)

Works as you described ✔️ 

In which case merging + a follow-up ticket for the counts sounds like the thing to do 🙌